osu.Game/Overlays/Rankings/CountryPill.cs

```diff
@@ -0,0 +1,149 @@
+            AutoSizeEasing = Easing.OutQuint;
+            Height = 25;
+            Masking = true;
+            Children = new Drawable[]
+            {
+                background = new Box
+                {
+                    RelativeSizeAxes = Axes.Both
+                },
+                new FillFlowContainer
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    RelativeSizeAxes = Axes.Y,
+                    AutoSizeAxes = Axes.X,
+                    Margin = new MarginPadding { Horizontal = 10 },
+                    Direction = FillDirection.Horizontal,
+                    Spacing = new Vector2(8, 0),
+                    Children = new Drawable[]
+                    {
+                        new FillFlowContainer
+                        {
+                            RelativeSizeAxes = Axes.Y,
+                            AutoSizeAxes = Axes.X,
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            Direction = FillDirection.Horizontal,
+                            Spacing = new Vector2(3, 0),
+                            Children = new Drawable[]
+                            {
+                                flag = new UpdateableFlag
+                                {
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                    Size = new Vector2(22, 15)
+                                },
+                                countryName = new OsuSpriteText
+                                {
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                    Font = OsuFont.GetFont(size: 14)
+                                }
+                            }
+                        },
+                        new CloseButton
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            Action = () => Country.Value = null
+                        }
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            background.Colour = colours.GreySeafoamDarker;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            Country.BindValueChanged(onCountryChanged, true);
+        }
+
+        public void Expand()
+        {
+            ClearTransforms();
+            AutoSizeAxes = Axes.X;
+            this.FadeIn(duration, Easing.OutQuint);
+        }
+
+        public void Collapse()
+        {
+            ClearTransforms();
+            AutoSizeAxes = Axes.None;
+            this.ResizeWidthTo(0, duration, Easing.OutQuint);
```

Comment:
Is this necessary? It doesn't seem to be to me, and it ticks me off how this never gets restored in `Expand()`.

Reply:
It's restored by itself when setting `AutosizeAxes = Axes.X` including the fact that we have `AutosizeDuration` and `AutosizeEasing` in the ctor.
We already have some UI components with similar logic (`PlayerSettings` for example)

Reply:
If by "necessary" you mean this Collapse/Expand animation then no, it's not. I was just thinking about what kind of animation we can apply here. Result seems pretty good to me, though.

Reply:
No, I just meant the resize call. Unless I screwed something up I didn't see any visual difference after removing it.

Reply:
I just checked: there's a difference after removing it. May be it's not really noticeable.